drivers/MbedCRC.h

```diff
@@ -0,0 +1,415 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_CRC_API_H
+#define MBED_CRC_API_H
+
+#include <stdint.h>
+#include "drivers/TableCRC.h"
+#include "platform/mbed_assert.h"
+
+#if defined ( __CC_ARM )
```

Comment:
Cant find a reference for these lines here. why push/pop is gere for GCC and that negative shift warning ignored.

Reply:
I am disabling warnings reported as "Shift count is negative", which is kind of invalid warning.
Warning is from below code section.
https://github.com/ARMmbed/mbed-os/pull/5911/files/af0982295f010f3dd45ade36ddc9155a5f86289e#diff-a353a07f9612be80e818a21f61ec8409R210

```
if ((width < 8) && (NULL == _crc_table)) {
            p_crc = (uint32_t)(p_crc << (8 - width));
 }
```
Compiler warns of the shift operation with width as it is width=(std::uint8_t), but we check for ( width < 8) before performing shift, so it should not be an issue.

Reply:
That is useful to know, please leave either in the code short description or commit message.